src_c/geometry.c

```diff
@@ -8,6 +8,108 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    double start_x, start_y;
+    double end_x, end_y;
+    PyObject **farr;
+    Py_ssize_t col_length;
+    Py_ssize_t loop;
+    double angle;
+    double max_dist;
+
+    if (nargs == 3) {
+        if (!PySequence_Check(args[0]) || !PySequence_Check(args[1]) || !PySequence_FAST_CHECK(args[2])) {
+            return RAISE(PyExc_TypeError,
+                        "argument of raycast() must be a sequence");
+        }
+        
+        if (!pg_TwoDoublesFromObj(args[0], &start_x, &start_y)) {
+            return RAISE(PyExc_TypeError, "the starting position requires a pair of floats");
+        }
+        if (!pg_TwoDoublesFromObj(args[1], &end_x, &end_y)) {
+            return RAISE(PyExc_TypeError, "the end position requires a pair of floats");
+        }
+        farr = PySequence_Fast_ITEMS(args[2]);
+        col_length = PySequence_Fast_GET_SIZE(args[2]);
+    }
+    else if (nargs == 4) {
+        if (!pg_TwoDoublesFromObj(args[0], &start_x, &start_y)) {
+            return RAISE(PyExc_TypeError, "the starting position requires a pair of floats");
+        }
+        if (!PyFloat_Check(args[1]) && !pg_DoubleFromObj(args[1], &angle)) {
+            return RAISE(PyExc_TypeError, "invalid angle");
+        }
+        if (!PyFloat_Check(args[2]) && !pg_DoubleFromObj(args[2], &max_dist)) {
+            return RAISE(PyExc_TypeError, "invalid max distance");
+        }
+        end_x = start_x - cos(angle * PI / 180) * max_dist;
+        end_y = start_y - sin(angle * PI / 180) * max_dist;
+        farr = PySequence_Fast_ITEMS(args[3]);
+        col_length = PySequence_Fast_GET_SIZE(args[3]);
+    }
+    else {
+        return RAISE(PyExc_TypeError,
+                    "invalid number of arguments");
+    }
+
+
+    pgLineObject *line = (pgLineObject *)pgLine_Type.tp_new(&pgLine_Type, NULL, NULL);
```

Comment:
You don-t need to create a whole python object, just create a pgLineBase struct here
```suggestion
    pgLineBase line;
```

Reply:
And from here on you should remove all `line->` (s)